tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -189,9 +185,8 @@ def test_upgrade_downgrade_brokers(self, from_version, to_version):
         processor.stop()
         processor.node.account.ssh_capture("grep SMOKE-TEST-CLIENT-CLOSED %s" % processor.STDOUT_FILE, allow_fail=False)
 
-    @matrix(from_version=metadata_1_versions, to_version=backward_compatible_metadata_2_versions)
-    @matrix(from_version=metadata_1_versions, to_version=metadata_3_or_higher_versions)
-    @matrix(from_version=metadata_2_versions, to_version=metadata_3_or_higher_versions)
+    @matrix(from_version=metadata_1_versions, to_version=[str(DEV_VERSION)])
+    @matrix(from_version=metadata_2_versions, to_version=[str(DEV_VERSION)])
```

Comment:
I was originally thinking that `streams_application_upgrade_test.py` and `streams_upgrade_test.py` can be re-consolidated so that the latter would be for broker-clients compatibility (including rolling upgrade the brokers), and would for now be ignored. While the former would be used for upgrading / downgrading clients for different paths (one rolling bounce including version probing, two rolling bounces, full shutdown and restart). In that case, this test and the test_version_probing_upgrade below could be moved into the other file. WDYT?

Reply:
Ah, I see.

Since you mention the client-broker test, I was going to pose this question to you: The test is currently ignored, as has been ignored for as long as I can remember. It seems like we should fix it or delete it. I'm guessing your preference would be to fix it, right?

To the point about the client-client metadata upgrade test, I left it and the version probing test here because they are specifically testing that the relevant metadata versions get upgraded, whereas the test I moved was instead just about making sure that the app keeps running after an upgrade.

It does seem like we could improve the metadata and broker upgrade tests by making them run the full soak app instead, but I'm wondering if we could do it separately. This PR in particular seems like a good addition to add asap, since it should dramatically reduce the number of tests we have to run during system tests.

Reply:
Yes I agree. It is not a `require` comment, I'm just leaving it since we've been discussing about merging various tests, especially even considering merging the StreamsSmokeTest with StreamsUpgradeTest but that can be done later.